### PR TITLE
Warn when using panic-strategy abort for proc-macro crates

### PR DESCRIPTION
**Pull request by Veykril**
See https://github.com/rust-lang/rust/issues/82320, this simply warns for now as that seems like the best step that can be immediately taken (opposed to straight up rejecting or ignoring)

### CODE REVIEW

**Comment by Veykril** [comment]
This seems to affect some other tests, is `abort` the default strategy for ui tests?

File: compiler/rustc_interface/src/passes.rs [hunk 1]
```diff
@@ -380,6 +381,10 @@ pub fn configure_and_expand(
         }
     }
 
+    if is_proc_macro_crate {
+        sess.emit_warning(ProcMacroCratePanicAbort);
+    }
```
[the remark this answers is not in the document]
Comment (Veykril) [reply compiler/rustc_interface/src/passes.rs:386]: It seems I forgot to add that change to the commit🤦

File: compiler/rustc_error_messages/locales/en-US/interface.ftl [hunk 2]
```diff
@@ -41,3 +41,6 @@ interface_rustc_error_unexpected_annotation =
 
 interface_failed_writing_file =
     failed to write file {$path}: {$error}"
+
+interface_proc_macro_crate_panic_abort =
+    building proc macro crate with `panic=abort` may crash the compiler should the proc-macro panic
```
Comment (eholk) [inline compiler/rustc_error_messages/locales/en-US/interface.ftl:46]:
```suggestion
    building a proc macro crate with `panic=abort` may crash the compiler should the proc-macro panic
```

**Comment by eholk** [comment]
Seems good to me. I made a tiny wording suggestion but feel free to r=me with or without the change.

File: compiler/rustc_error_messages/locales/en-US/interface.ftl (hunk 2 above)
Comment (Veykril) [reply compiler/rustc_error_messages/locales/en-US/interface.ftl:46]: I used that wording due to the same being used for the following error/warning

> interface_proc_macro_doc_without_arg =
>    Trying to document proc macro crate without passing '--crate-type proc-macro to rustdoc
> .warn = The generated documentation may be incorrect

I am fine with changing this, but should that message there also be changed for consistency?

Comment (eholk) [reply compiler/rustc_error_messages/locales/en-US/interface.ftl:46]: I was kind of on the fence with this suggestion anyway, and I think consistency with the other messages is a good reason to stick with your original wording. Let's leave it as is.

**Review by albertlarsan68 — CHANGES_REQUESTED** [review]
Tests have moved from `src/test` to `tests`, please [rebase](https://rustc-dev-guide.rust-lang.org/git.html#rebasing) on top of new `master`.
@rustbot author
@bors r-

**Pull request merged**